frontend/src/main/scala/bloop/data/SettingsMapper.scala

```diff
@@ -0,0 +1,106 @@
+package bloop.data
+import bloop.engine.BuildLoader
+import bloop.logging.Logger
+import bloop.logging.DebugFilter
+import java.nio.charset.StandardCharsets
+import bloop.config.ConfigEncoderDecoders
+import io.circe.Printer
+import io.circe.Json
+import io.circe.parser
+import bloop.io.AbsolutePath
+import java.nio.file.Files
+import bloop.config.Config.Settings
+import io.circe.JsonObject
+import _root_.bloop.config.Config
+import bloop.DependencyResolution
+import scala.util.Try
+
+object SettingsMapper {
+
+  /**
+    * @param settingsPath Path to the json settings file.
+    * @param logger The logger that collects messages about loading the settings.
+    * @return Settings if the file exists.
+    */
+  def fromFile(settingsPath: AbsolutePath, logger: Logger): Option[Settings] = {
+    if (settingsPath.isFile) {
+      val bytes = Files.readAllBytes(settingsPath.underlying)
+      logger.debug(s"Loading workspace settings from ${BuildLoader.settingsFileName}")(
+        DebugFilter.All
+      )
+      val contents = new String(bytes, StandardCharsets.UTF_8)
+      parser.parse(contents) match {
+        case Left(failure) => throw failure
+        case Right(json) => Some(fromJson(json))
+      }
+    } else {
+      None
+    }
+  }
+
+    /**
+    * @param json Json object to extract settings from.
+    * @return Settings extracted from the json.
+    */
+  def fromJson(json: Json): Settings = {
+    ConfigEncoderDecoders.settingsDecoder.decodeJson(json) match {
+      case Right(settings) => settings
+      case Left(failure) => throw failure
+    }
+  }
+
+    /**
+    * @param target Path to where the json settings file should be saved.
+    * @param settings The settings to save.
+    */
+  def save(target: AbsolutePath, settings: Settings): Unit = {
```

Comment:
Added Either, but in most places if there is an error, we just ignore it - since there is not much we can do if writing fails. We could just log an error, does it make sense?